File: api/Backend/CCQuartersAPI/Requests/UpdateUserRequest.cs
using CCQuartersAPI.CommonClasses;

namespace CCQuartersAPI.Requests
{
    public class UpdateUserRequest
    {
        public string? Name { get; set; }
        public string? Surname { get; set; }
        public string? Company { get; set; }
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
    }
}

File: client/lib/services/users/requests/update_user_request.dart
import 'package:ccquarters/model/user/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'update_user_request.g.dart';

@JsonSerializable()
class UpdateUserRequest {
  UpdateUserRequest(
      {required this.name,
      required this.surname,
      this.company,
      required this.email,
      this.phoneNumber});

  String? name;
  String? surname;
  String? company;
  String email;
  String? phoneNumber;

  UpdateUserRequest.fromUser(User user)
      : name = user.name,
        surname = user.surname,
        company = user.company,
        email = user.email,
        phoneNumber = user.phoneNumber;

  Map<String, dynamic> toJson() => _$UpdateUserRequestToJson(this);
  factory UpdateUserRequest.fromJson(Map<String, dynamic> json) =>
      _$UpdateUserRequestFromJson(json);
}
